packager.go

```diff
@@ -252,6 +275,10 @@ func (this Packager) fetchURL(orig *url.URL) (*http.Request, *http.Response, *HT
 		return nil, nil, NewHTTPError(http.StatusInternalServerError, "Error building request: ", err)
 	}
 	req.Header.Set("User-Agent", userAgent)
+	// Set conditional headers that were included in ServeHTTP's Request.
+	for header := range conditionalRequestHeaders {
+		req.Header.Set(header, serveHTTPReq.Get(header))
```

Comment:
Wrap this in `if serveHTTPReq.Get(header) != ""`; otherwise it sends empty headers for each of these.

Reply:
Ah, this one was in that newer commit. Thank you for the thoroughness.